tsdb/shard.go

```diff
@@ -53,6 +53,9 @@ type Shard struct {
 	walPath string
 	id      uint64
 
+	database        string
```

Comment:
Why not store the `ShardLocation` instead? Wouldn't need to explicitly store `walPath` or `path` either.